File: seminar/src/main/java/com/server/dosopt/seminar/domain/ServiceMember.java
package com.server.dosopt.seminar.domain;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
public class ServiceMember {

    @Id
    @GeneratedValue
    private Long id;

    private String nickanme;
    private String password;

    @Builder
    public ServiceMember(String nickanme, String password) {
        this.nickanme = nickanme;
        this.password = password;
    }
}

File: kotlin-seminar/src/main/kotlin/sopt/org/kotlinseminar/domain/ServiceMember.kt
package sopt.org.kotlinseminar.domain

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id

@Entity
class ServiceMember(
        nickname: String,
        password: String,

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        val id: Long = 0L
) {

    @Column(nullable = false)
    var nickname = nickname
        protected set

    @Column(nullable = false)
    var password = password
        protected set
}